code/modules/mining/lavaland/ruins/dwarfruin.dm

```diff
@@ -0,0 +1,103 @@
+#define DWARF_COOLDOWN 6000 //10 minutes
+GLOBAL_LIST_INIT(dwarves_list, list())
+
+/*Dwarf Spawner*/
+/obj/machinery/migrant_spawner
+	name = "strange portal"
+	desc = "A strange portal that acts as a gateway between this world and another. Strange pixelated images fade in and out inside the frame."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "dorfportal"
+	anchored = TRUE
+	var/respawn_cooldown = DWARF_COOLDOWN
+	var/list/spawned_dorfs = list()
+	var/list/spawned_mobs = list()
+	var/list/recently_dead_ckeys = list()
+	var/dorf_gear = /datum/outfit/dorf
+
+/obj/machinery/migrant_spawner/Initialize()
+	. = ..()
+	GLOB.poi_list += src
+
+/obj/machinery/migrant_spawner/Destroy()
+	GLOB.poi_list.Remove(src)
+	return ..()
+
+/obj/machinery/migrant_spawner/process()
+	for(var/i in spawned_mobs)
+		if(!i)
+			LAZYREMOVE(spawned_mobs, i)
+			continue
+		var/mob/living/M = i
+		if(M.stat == DEAD || !M.key) //adding timer to dead/ghosted dwarfs
+			add_dorf_timer(M)
+
+/obj/machinery/migrant_spawner/attack_ghost(mob/user)
+	if(!SSticker.HasRoundStarted())
+		return
+	if(user.ckey in spawned_dorfs)
+		if(user.ckey in recently_dead_ckeys)
+			to_chat(user, "It must be more than [respawn_cooldown/600] minutes from your last death to respawn!")
```

Comment:
It does not apply to multiple people spawning in as dwarves so effectively providing you don't die there is no restriction on spawning.